### DISCUSSION TITLE
setting a dynamic row height for each row of a table

### DISCUSSION THREAD

**Discussion by Nithyashri-E** [py-pdf/fpdf2#818] [Q&A]
|------------------------------|------------------------------------------|
|------------------------------|------------------------------------------|               
|                                            |                                                             |                   
|                                            |                                                             |                     
|------------------------------|------------------------------------------|
|                                            |                                                             |
|------------------------------|------------------------------------------|

Let's name the above table's cell as follows:
| cell1       |        cell2                   |
|  cell3      |         cell4                  |
| cell5       |         cell6                  |

A string of words with varying lengths goes into cell1, cell3, cell5. My requirement is to make these cell's row height auto adjust based upon its string length and the font size considered.

NOTE: The row height used for cell1 must be used for cell2 too and the row height used in cell3 must be used for cell4 and so on. Is there any fpdf method I can use to do this in python?

**Comment by Lucas-C** [comment]
Hi @Nithyashri-E

This is not currently doable, but this is an interesting feature suggestion!

A potential approach could be
1. to allow a value of "AUTO" to the parameter `col_widths` passed to `FPDF.table()`
2. add `MethodReturnValue.LINES` to the value of `output` passed to `FPDF.multi_cell()` in `Table._render_table_cell()`, in order to be able to pre-compute line lengths
3. refactor the code to call `_get_row_layout_info()` for **every line** at the beginning of `Table.render()`, so that we can measure the average length of every table column
4. finally, in `Table.render()`, if `self._col_widths == "AUTO"`, use this this everage length per column as widths

**Reply by Lucas-C** [reply]
Woops, the solution I suggested would allow to implement **automatic column widths**

But you wished for **automatic row heights**! Sorry, I mixed up things 😅

Anyway, thank you for reaching out 😊
**Automatic row heights** is already implemented and should work!
Hence, could you please provide some minimal code that reproduce your issue?
